src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -219,70 +218,75 @@ export const handleGetPublicForm: ControllerHandler<
-                },
-                isIntranetUser,
-              })
-          })
-          .mapErr((error) => {
-            // NOTE: If the user is not signed in or if the user refreshes the page while logged in, it is not an error.
-            // myInfoError is set to true only when the authentication provider rejects the user's attempt at auth
-            // or when there is a network or database error during the process of retrieval
-            const isMyInfoError = !(
-              error instanceof MyInfoCookieAccessError ||
-              error instanceof MyInfoMissingAccessTokenError
+      return extractAuthCode(authCodeCookie)
+        .asyncAndThen((authCode) => MyInfoService.retrieveAccessToken(authCode))
+        .andThen((accessToken) =>
+          MyInfoService.getMyInfoDataForForm(form, accessToken),
+        )
+        .andThen((myInfoData) =>
+          BillingService.recordLoginByForm(form).map(() => myInfoData),
+        )
+        .andThen((myInfoData) => {
+          return MyInfoService.prefillAndSaveMyInfoFields(
+            form._id,
+            myInfoData,
+            form.toJSON().form_fields,
+          ).map((prefilledFields) => ({
+            prefilledFields,
+            spcpSession: { userName: myInfoData.getUinFin() },
+            myInfoLoginCookie: createMyInfoLoginCookie(myInfoData.getUinFin()),
+          }))
+        })
+        .map(({ myInfoLoginCookie, prefilledFields, spcpSession }) => {
+          // Set the updated cookie accordingly and return the form back to the user
+          return res
+            .cookie(
+              MYINFO_LOGIN_COOKIE_NAME,
+              myInfoLoginCookie,
+              MYINFO_LOGIN_COOKIE_OPTIONS,
             )
-            // No need for cookie if data could not be retrieved
-            // NOTE: If the user does not have any cookie, clearing the cookie still has the same result
-            return res
-              .clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
-              .json({
-                form: publicForm,
-                // Setting to undefined ensures that the frontend does not get myInfoError if it is false
-                myInfoError: isMyInfoError || undefined,
-                isIntranetUser,
-              })
+            .json({
+              spcpSession,
+              form: {
+                ...publicForm,
+                form_fields: prefilledFields as FormFieldDto[],
+              },
+              isIntranetUser,
+            })
+        })
+        .mapErr((error) => {
+          logger.error({
+            message: 'MyInfo login error',
+            meta: logMeta,
+            error,
           })
-      )
+          // No need for cookie if data could not be retrieved
+          // NOTE: If the user does not have any cookie, clearing the cookie still has the same result
+          return res
+            .clearCookie(MYINFO_LOGIN_COOKIE_NAME, MYINFO_LOGIN_COOKIE_OPTIONS)
```

Comment:
this has already been cleared above in line 223, should we remove it here?

Reply:
good catch, removed

Reply:
676072bc